java/compression/src/main/java/org/apache/arrow/compression/CommonsCompressionFactory.java

```diff
@@ -27,15 +27,22 @@
  */
 public class CommonsCompressionFactory implements CompressionCodec.Factory {
 
-  public static final CommonsCompressionFactory INSTANCE = new CommonsCompressionFactory();
+  private int compressionLevel;
+
+  public CommonsCompressionFactory(int compressionLevel) {
+    this.compressionLevel = compressionLevel;
+  }
+
+  @Deprecated
+  public static final CommonsCompressionFactory INSTANCE = new CommonsCompressionFactory(3);
 
   @Override
   public CompressionCodec createCodec(CompressionUtil.CodecType codecType) {
```

Comment:
Is it better to extend `createCodec` to add an `Optional<int> level`? The downside of the current implementation is that we need to create separate instances for different levels.

Reply:
Just updated.

```
ArrowFileWriter writer = new ArrowFileWriter(vectorSchemaRoot, null,
  fileOutputStream.getChannel(), null, IpcOption.DEFAULT,
  CommonsCompressionFactory.INSTANCE, zstdCodec.getCodecType(), /*compressionLevel*/ 7)
```

Reply:
I'm going to fill a ticket to implement Builder patter for Reader/Writes class.